src/games/erdos_renyi.c

```diff
@@ -244,6 +244,49 @@ igraph_error_t igraph_erdos_renyi_game_gnm(
     return IGRAPH_SUCCESS;
 }
 
+igraph_error_t igraph_erdos_renyi_game_gnm_multi(
+    igraph_t *graph, igraph_integer_t n, igraph_integer_t m,
+    igraph_bool_t directed, igraph_bool_t loops
+) {
+
+    igraph_integer_t no_of_nodes = n;
+    igraph_integer_t no_of_edges = m;
+    igraph_vector_int_t edges;
+
+    if (n < 0) {
+        IGRAPH_ERROR("Invalid number of vertices.", IGRAPH_EINVAL);
+    }
+    if (m < 0 || m > IGRAPH_ECOUNT_MAX) {
+        IGRAPH_ERROR("Invalid number of edges.", IGRAPH_EINVAL);
+    }
+
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&edges, 0);
+    IGRAPH_CHECK(igraph_vector_int_reserve(&edges, no_of_edges * 2));
+
+    RNG_BEGIN();
+    for (igraph_integer_t i = 0; i < no_of_edges; i++) {
```

Comment:
It's great that you localized the `i` variable to the `for` loop. Please keep doing this in the future even though this is not what you can currently see in a lot of old igraph code.